src/AggregateFunctions/AggregateFunctionStatisticsSimple.h

```diff
@@ -277,4 +282,140 @@ template <typename T1, typename T2> using AggregateFunctionCovarPopSimple = Aggr
 template <typename T1, typename T2> using AggregateFunctionCovarSampSimple = AggregateFunctionVarianceSimple<StatFuncTwoArg<T1, T2, StatisticsFunctionKind::covarSamp>>;
 template <typename T1, typename T2> using AggregateFunctionCorrSimple = AggregateFunctionVarianceSimple<StatFuncTwoArg<T1, T2, StatisticsFunctionKind::corr>>;
 
+
+template <StatisticsFunctionKind _kind>
+struct StatFuncArbitraryArgData
+{
+    using DataType = std::conditional_t<_kind == StatisticsFunctionKind::corr, CorrMoments<Float64>, CovarMoments<Float64>>;
```

Comment:
If both data type of the arguments are `Float32`, then result type is better be `Float32`.

Reply:
Forwarding all arguments types to template is a problem.